public/chat-client.html

```diff
@@ -0,0 +1,216 @@
+
+      const fillPrevChats = () => {
+        fetch('https://script-heroes-bn-backend-stgng.herokuapp.com/api/chat', {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            'x-auth-token': token
+          }
+        })
+          .then((res) => {
+            if (res.status !== 200) {
+              loginSection.classList.remove('hide');
+              chatSection.classList.add('hide');
+            }
+            return res.json();
+          })
+          .then((res) => {
+            console.log(res);
+            if (res.data?.length) {
+              console.log(res.data);
+              res.data.map((msg) => {
+                displayMessage(msg);
+              });
+            }
+          })
+          .catch((err) => console.log(err));
+      };
+
+      checkAuth();
+      fillPrevChats();
+      const loginForm = document.querySelector('#login-form form');
+      loginForm.addEventListener('submit', (e) => {
+        e.preventDefault();
+        const email = loginForm.querySelector('#email');
+        const password = loginForm.querySelector('#password');
+
+        fetch('https://script-heroes-bn-backend-stgng.herokuapp.com/api/auth/login', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            email: email.value,
+            password: password.value
+          })
+        })
+          .then((res) => res.json())
+          .then((data) => {
+            token = data?.token;
+            localStorage.setItem('token', data?.token);
+            if (checkAuth()) {
+              fillPrevChats();
+            }
+          })
+          .catch((err) => console.log(err));
+      });
+
+      const chatForm = document.querySelector('#chat-room form');
+      const chatInput = chatForm.querySelector('input');
+      const sendButton = chatForm.querySelector('button');
+
+      chatInput.addEventListener('keyup', (e) => {
+        if (!e.target.value) {
+          sendButton.disabled = true;
+        } else {
+          sendButton.disabled = false;
+        }
+      });
+
+      chatForm.addEventListener('submit', (e) => {
+        e.preventDefault();
+        fetch('https://script-heroes-bn-backend-stgng.herokuapp.com/api/chat', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            'x-auth-token': token
+          },
+          body: JSON.stringify({
+            message: chatInput.value
+          })
+        })
+          .then((res) => res.json())
+          .then((data) => {
+            chatInput.value = '';
+            sendButton.disabled = true;
+          })
+          .catch((err) => console.log(err));
+      });
+
+      var socket = io.connect('localhost:3000');
+      socket.on('message', (msg) => {
+        console.log(msg);
+        displayMessage(msg);
+      });
+    </script>
+  </body>
+</html>
```

Comment:
could you please not include this one to the codebase, since it won't be used much. Maybe you could do it aside for demo purposes

Reply:
Fixed 👍